bigbio/utils/license.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+License objects.
+"""
+
+import importlib.resources as pkg_resources
+import json
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+from bigbio.utils import resources
+
+
+@dataclass
+class License:
+    """
+    Base class from which all licenses inherit
+    
+    Args:
+        name: License title
+        text: Accompanying information of the license
+        link: URL to License
+        version: Current version of license
+        provenance: Organization providing authorization, if possible
+    """
+
+    name: Optional[str] = None
+    text: Optional[str] = None
+    link: Optional[str] = None
+    version: Optional[float] = None
+    provenance: Optional[str] = None
+
+    @property
+    def is_share_alike(self):
+        """
+        Is Share-alike?
+        """
+
+        return False
```

Comment:
before merging this can we just raise a not implemented error here or remove?

Reply:
I added a `NotImplementedError`, I like to have it there just to showcase what it is possible to do in case somebody else will work w/ this.